lockedFile.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"os"
+	"syscall"
+)
+
+// lockedFile is a wrapper around `os.File` which overrides the `Open` and `Close` methods so that the file being
+// accessed is exclusively locked (via `flock(..., LOCK_EX)`).
```

Comment:
```suggestion
// lockedFile is a wrapper around `os.File` that adds `.Open()` and overrides `.Close()` methods so that the
// underlying file is exclusively locked (via `flock(..., LOCK_EX)`) when accessed.
```